python-threatexchange/threatexchange/fetcher/fetch_api.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+The fetcher is the component that talks to external APIs to get and put signals
+
+@see Fetcher
+"""
+
+
+from dataclasses import dataclass
+from enum import Enum
+import typing as t
+
+from threatexchange.collab_config import CollaborationConfig
+from threatexchange.signal_type.signal_base import SignalType
+from . import fetch_state_base as state
+
+
+class SignalExchangeAPI:
+    """
+    An external APIs to get and maybe put signals
+
+    Fetchers ideally can checkpoint their progress, so that they can tail
+    updates from the server.
+
+    Is assumed that fetched signals have some metadata attached to them,
+    which is unique to that API. Additionally, it a assumed that there
+    might be multiple contributors to signals inside of an API.
+
+    The Fetcher can assume it is stateful, and
+    """
+
+    @classmethod
+    def name(cls) -> str:
+        """A human-friendly name unique to all fetchers"""
+        return cls.__name__
+
+    def resolve_owner(self, id: int) -> str:
+        """Convert an owner ID into an owner record"""
+        return ""
```

Comment:
One of my earlier drafts relied on this more, but I think I'll keep it in for now, since I think this is where that name resolution would happen.

Reply:
can we have it "raise NotImplementedError" to it does fail silently/non-uniquely

Reply:
I'd imagined this being used only by default for human readable stuff, so I think it's okay to leave this way.